conductor-client/src/main/kotlin/com/openlattice/postgres/external/ExternalDatabasePermissioner.kt

```diff
@@ -0,0 +1,515 @@
+            val roleIds = roles.map { PostgresRoles.buildPostgresRoleName(it.id) }
+            val roleIdsSql = roleIds.joinToString { DataTables.quote(it) }
+
+            dataSource.connection.use { connection ->
+                connection.createStatement().use { statement ->
+
+                    logger.info("Revoking ${Schemas.PUBLIC_SCHEMA} schema right from roles: {}", roleIds)
+                    //Don't allow users to access public schema which will contain foreign data wrapper tables.
+                    statement.execute("REVOKE USAGE ON SCHEMA ${Schemas.PUBLIC_SCHEMA} FROM $roleIdsSql")
+                }
+            }
+        }
+    }
+
+    override fun createRole(role: Role) {
+        val (dbRole, _) = dbCredentialService.getOrCreateRoleAccount(role)
+
+        logger.debug("Creating role if not exists {}", dbRole)
+        atlas.connection.use { connection ->
+            connection.createStatement().use { statement ->
+                statement.execute(createRoleIfNotExistsSql(dbRole))
+                //Don't allow users to access public schema which will contain foreign data wrapper tables.
+                logger.debug("Revoking ${Schemas.PUBLIC_SCHEMA} schema right from role: {}", role)
+                statement.execute("REVOKE USAGE ON SCHEMA ${Schemas.PUBLIC_SCHEMA} FROM ${DataTables.quote(dbRole)}")
+            }
+        }
+    }
+
+    override fun createUnprivilegedUser(principal: SecurablePrincipal) {
+        /**
+         * To simplify work-around for ESRI username limitations, we are only introducing one additional
+         * field into the dbcreds table. We keep the results of calling [buildPostgresUsername] as the lookup
+         * key, but instead use the username and password returned from the db credential service.
+         */
+        val (dbUser, dbUserPassword) = dbCredentialService.getOrCreateDbAccount(principal)
+
+        logger.debug("Creating user if not exists {}", dbUser)
+        atlas.connection.use { connection ->
+            connection.createStatement().use { statement ->
+                //TODO: Go through every database and for old users clean them out.
+//                    logger.info("Attempting to drop owned by old name {}", user.name)
+//                    statement.execute(dropOwnedIfExistsSql(user.name))
+//                    logger.info("Attempting to drop user {}", user.name)
+//                    statement.execute(dropUserIfExistsSql(user.name)) //Clean out the old users.
+//                    logger.info("Creating new user {}", dbUser)
+                statement.execute(createUserIfNotExistsSql(dbUser, dbUserPassword))
+                //Don't allow users to access public schema which will contain foreign data wrapper tables.
+                logger.debug("Revoking ${Schemas.PUBLIC_SCHEMA} schema right from user {}", principal)
+                statement.execute("REVOKE USAGE ON SCHEMA ${Schemas.PUBLIC_SCHEMA} FROM ${DataTables.quote(dbUser)}")
```

Comment:
might be worth extracting the actual sql strings into constants/functions

Reply:
Definitely worth extracting but just ported from ACM. Can do as follow-up